### PR TITLE
ADM-744:[frontend]feat: add board verify failed popup

### PR DESCRIPTION
**Pull request by weiraneve**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: frontend/e2e/pages/metrics/report-step.ts [hunk 1]
```diff
@@ -230,16 +230,16 @@ export class ReportStep {
     await expect(this.classificationRows.nth(10)).toContainText(this.combineStrings(['1.0', '88.89%']));
     await expect(this.classificationRows.nth(11)).toContainText(this.combineStrings(['None', '11.11%']));
     await expect(this.classificationRows.nth(13)).toContainText(this.combineStrings(['Auto Dora Metrics', '100.00%']));
+    await expect(this.classificationRows.nth(14)).toContainText(this.combineStrings(['Sprint']));
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/report-step.ts:233]: 没有value？

File: frontend/e2e/pages/metrics/report-step.ts [hunk 2]
```diff
@@ -230,16 +230,16 @@ export class ReportStep {
...
-    await expect(this.classificationRows.nth(16)).toContainText(this.combineStrings(['Sprint 27', '100.00%']));
     await expect(this.classificationRows.nth(17)).toContainText(this.combineStrings(['Sprint 28', '88.89%']));
     await expect(this.classificationRows.nth(19)).toContainText(this.combineStrings(['None', '100.00%']));
     await expect(this.classificationRows.nth(21)).toContainText(this.combineStrings(['None', '100.00%']));
     await expect(this.classificationRows.nth(23)).toContainText(this.combineStrings(['Medium', '100.00%']));
     await expect(this.classificationRows.nth(25)).toContainText(this.combineStrings(['None', '100.00%']));
     await expect(this.classificationRows.nth(27)).toContainText(this.combineStrings(['None', '100.00%']));
-    await expect(this.classificationRows.nth(29)).toContainText(this.combineStrings(['Stream1', '44.44%']));
-    await expect(this.classificationRows.nth(30)).toContainText(this.combineStrings(['Stream2', '55.56%']));
+    await expect(this.classificationRows.nth(28)).toContainText(this.combineStrings(['Labels']));
```
Comment (guzhongren) [inline frontend/e2e/pages/metrics/report-step.ts:241]: 没有value？

File: frontend/src/containers/ConfigStep/FormAlert/index.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        </EllipsisText>
+      );
+    }
+  };
+
+  return (
+    <>
+      {showAlert && (
+        <StyledAlert
+          data-testid={FormAlertDataTestIdMap(formAlertType)}
```
Comment (guzhongren) [inline frontend/src/containers/ConfigStep/FormAlert/index.tsx:46]: 不是说不让用testid了么

**Review by guzhongren — COMMENTED** [review]
change

**Comment by guzhongren** [comment]
我们项目前端的选择器之前定下来的共识是建议用aria-label=‘natual language’，内容就是自然语言（而不是驼峰或蛇形），以便同时达到getByLabelText和accessibility的目的，所以今后请大家都写aria-label而不是data-test-id；抱歉这个是我没有及时同步\ slack

File: frontend/e2e/pages/metrics/report-step.ts (hunk 2 above)
Comment (weiraneve) [reply frontend/e2e/pages/metrics/report-step.ts:241]: 我再check和fix下

File: frontend/e2e/pages/metrics/config-step.ts [hunk 4]
```diff
@@ -145,6 +146,9 @@ export class ConfigStep {
     this.boardTokenErrorMessage = this.boardContainer.getByText(
       'Token is invalid, please change your token with correct access permission!',
     );
+    this.boardVerifyFailedPopupMessage = this.boardContainer.getByText(
+      'Email and Token are bound for verification. Please modify at least one of the Email or Token before reverify!',
```
[the remark this answers is not in the document]
Comment (weiraneve) [reply frontend/e2e/pages/metrics/config-step.ts:150]: 这是按照figma的设计来做的

File: frontend/src/containers/ConfigStep/FormAlert/index.tsx (hunk 3 above)
Comment (weiraneve) [reply frontend/src/containers/ConfigStep/FormAlert/index.tsx:46]: 这里我们code review的意见是暂时先用着data test id，因为这里涉及到的重构的组件之前对应的测试改动很多

**Review by zhou-yinyuan — APPROVED** [review]

File: frontend/src/containers/ConfigStep/BoardVerifyAlert/style.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { styled } from '@mui/material/styles';
+import { Alert } from '@mui/material';
+import { theme } from '@src/theme';
+
+export const StyledAlert = styled(Alert)({
+  '&.MuiPaper-root': {
```
Comment (neomgb) [inline frontend/src/containers/ConfigStep/BoardVerifyAlert/style.tsx:6]: space？

File: frontend/src/containers/ConfigStep/FormAlert/style.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { styled } from '@mui/material/styles';
+import { Alert } from '@mui/material';
+import { theme } from '@src/theme';
+
+export const StyledAlert = styled(Alert)({
+  '&.MuiPaper-root': {
```
Comment (neomgb) [inline frontend/src/containers/ConfigStep/FormAlert/style.tsx:6]: 同上

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx [hunk 7]
```diff
@@ -208,6 +209,32 @@ describe('Board', () => {
...
+    const mockedError = new UnauthorizedError('', HttpStatusCode.Unauthorized, '');
+    boardClient.getVerifyBoard = jest.fn().mockImplementation(() => Promise.reject(mockedError));
+
+    await userEvent.click(screen.getByText(VERIFY));
+
+    expect(screen.getByTestId('boardVerifyAlert')).toBeInTheDocument();
+
+    await userEvent.click(screen.getByLabelText('Close'));
+
+    expect(screen.queryByLabelText('boardVerifyAlert')).not.toBeInTheDocument();
```
Comment (mrcuriosity-tw) [inline frontend/__tests__/containers/ConfigStep/Board.test.tsx:235]: line 231 uses data-test-id, while line 235 uses aria-label.
1. choose one pattern to use(we're currently using aria-label, which means you may need to refactor some `data-testid={FormAlertDataTestIdMap(formAlertType)}` in `FormAlert.tsx`.
2. Actually you didn't give an aria-label to the alert, so line 235 makes no sense here.

File: frontend/src/containers/ConfigStep/FormAlert/index.tsx (hunk 3 above)
Comment (mrcuriosity-tw) [reply frontend/src/containers/ConfigStep/FormAlert/index.tsx:46]: 用aria-label吧

**Review by neomgb — APPROVED** [review]

**Review by mrcuriosity-tw — COMMENTED** [review]
See `src/containers/ConfigStep/BoardVerifyAlert/index.tsx`, it's refferenced by no one, fogget to remove it?

File: frontend/__tests__/containers/ConfigStep/Board.test.tsx (hunk 7 above)
Comment (weiraneve) [reply frontend/__tests__/containers/ConfigStep/Board.test.tsx:235]: - board和timeout的alert代码已经被移除👌

- 对于data test id的重构是之前code review讨论的，如果改动的测试代码点过多，就暂且保留嘛

**Review by mrcuriosity-tw — APPROVED** [review]

**Pull request merged**